manticore/ethereum/manticore.py

```diff
@@ -1552,29 +1549,38 @@ def generate_testcase(self, state, message="", only_if=None, name="user"):
 
             m.generate_testcase(state, 'balance CAN be 0', only_if=balance == 0)
             # testcase generated with an input that will violate invariant (make balance == 0)
+        """
+        try:
+            with state as temp_state:
+                if only_if is not None:
+                    temp_state.constrain(only_if)
+                return self.generate_testcase_ex(temp_state, message, name=name)
+        except ManticoreError:
+            return None
+
+    def generate_testcase_ex(self, state, message="", name="user"):
+        """
+        Generate a testcase in the outputspace for the given program state.
+        An exception is raised if the state is unsound or unfeasible
+
+        On return you get a Testcase containing all the informations about the state.
+        A Testcase is a collection of files living at the OutputSpace.
+        Registered plugins and other parts of Manticore add files to the Testcase.
+        User can add more streams to is like this:
+
+        testcase = m.generate_testcase_ex(state)
+        with testcase.open_stream("txt") as descf:
+            descf.write("This testcase is interesting!")
 
         :param manticore.core.state.State state:
         :param str message: longer description of the testcase condition
-        :param manticore.core.smtlib.Bool only_if: only if this expr can be true, generate testcase. if is None, generate testcase unconditionally.
-        :param str name: short string used as the prefix for the workspace key (e.g. filename prefix for testcase files)
-        :return: If a testcase was generated
+        :param str name: short string used as the prefix for the outputspace key (e.g. filename prefix for testcase files)
+        :return: a Testcase
         :rtype: bool
         """
-        """
-        Create a serialized description of a given state.
-        :param state: The state to generate information about
-        :param message: Accompanying message
-        """
+        # Refuse to generate a testcase from an unsound state
         if not self.fix_unsound_symbolication(state):
-            return False
-
-        if only_if is not None:
```

Comment:
Removing this will change the `only_if` behavior that we discussed w/ @montyly on slack, yeah? If so, we should document that in the release notes.

Reply:
Supposedly nothing is changed from the user perspective.
Added `generate_testcase_ex`  and kept `generate_testcase` with the original behaivor.

The annoying-ish thing about the current behavior is that `mcoreEVM.generate_testcase`: 
It does a bunch of unexpected operations over a state mostly to filter out states that are not feasible. So it is more a "try_generate_testcase".  
Trying to do the same thing by hand via state.solve_xxxx would not give same result unless you replicate all the secret sauce of `mcoreEVM.generte_testcase`.

 `https://github.com/trailofbits/manticore/pull/1717/files#diff-e4836efb704ed46c30e5cfb02496a419R1538

Reply:
Oh my bad, this is an entirely different function now. Sounds good!